internal/backend/remote-state/s3/client.go

```diff
@@ -287,6 +288,26 @@ func (c *RemoteClient) Lock(info *statemgr.LockInfo) (string, error) {
 		info.ID = lockID
 	}
 
+	if err := c.s3Lock(info); err != nil {
+		return "", err
+	}
+	if err := c.dynamoDBLock(info); err != nil {
+		// when the second lock fails from getting acquired, release the initially acquired one
+		if uErr := c.s3Unlock(info.ID); uErr != nil {
+			log.Printf("[WARN] failed to release the S3 lock after failed to acquire the dynamoDD lock: %v", uErr)
```

Comment:
What do you think about using a multierror here?

Reply:
I attempted to do so in `Unlock()` as well, but the problem is that there are some places where do rely on `LockError` to be returned. One example of that would be [this one](https://github.com/opentofu/opentofu/blob/1012ccf8c91a12f75050a6b2a0e2c996efb05513/internal/states/statemgr/locker.go#L91-L96).
IMHO, there should have been used `errors.As` instead and then we can safely use multierror in this context too.

Reply:
Ah, yeah that's tricky as this codebase predates a lot of modern convenience.

Let's leave it alone for now.

Reply:
Should we create a ticket for getting into tackling this issue eventually?